### PR TITLE
Mobile: Remove v0 logic from appointments controller and tests

### PR DESCRIPTION
**Pull request by Tonksthebear**
## Summary
Removes v0 logic from the appointments controller and removes their corresponding tests. Renames a couple of functions that specify version number because we are no longer using multiple versions

## Related issue(s)
[department-of-veterans-affairs/va.gov-team#3430](https://app.zenhub.com/workspaces/va-mobile---backend-61d75d03a7505500147ec7f6/issues/gh/department-of-veterans-affairs/va-mobile-app/3430)

## Testing done

Removes tests that correspond to removed code

## What areas of the site does it impact?
None

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

## Requested Feedback

Are these all the areas that use v0 appointment code?

### CODE REVIEW

**Comment by Tonksthebear** [comment]
From what I can tell, there are some specs from the v0 that were not carried over one to one to the v2 spec. We may want to create a separate ticket for ensuring we're still testing all of those cases

**Review by kpethtel — CHANGES_REQUESTED** [review]
You should remove all references to the `mobile_appointment_use_VAOS_v2` flag since it will no longer be in use after this change.

**Comment by Tonksthebear** [comment]
This is obviously an insanely large PR. A large majority are removing/relocating cassettes. Not sure what the best way is to navigate such a large change

File: modules/mobile/config/initializers/statsd.rb [hunk 1]
```diff
@@ -24,15 +24,6 @@
 
   # Appointments #---------------------------------------------------------------
 
-  # meta binding
```
Comment (Tonksthebear) [inline modules/mobile/config/initializers/statsd.rb:27]: Should we change this to the V2 proxy or is this no longer necessary?

File: modules/mobile/app/services/mobile/v0/appointments/proxy.rb [hunk 2]
```diff
@@ -6,152 +6,17 @@ module Mobile
   module V0
     module Appointments
       class Proxy
```
Comment (Tonksthebear) [inline modules/mobile/app/services/mobile/v0/appointments/proxy.rb:8]: Should these couple of functions remain in the Proxy? Is seems to no longer serve my understanding of a Proxy, but I wasn't sure

File: modules/mobile/config/initializers/statsd.rb (hunk 1 above)
Comment (aherzberg) [reply modules/mobile/config/initializers/statsd.rb:27]: Ideally this should be tracked in datadog but right now appointments are lumped all into one metric. I know I'm the one that initially suggested to delete it but now I'm thinking we should change to v2 until it shows in datadog properly. if we end up deleting though we should delete the next 4 lines too as they initialize the metrics that are getting deleted.

File: modules/mobile/app/services/mobile/v0/appointments/proxy.rb (hunk 2 above)
Comment (aherzberg) [reply modules/mobile/app/services/mobile/v0/appointments/proxy.rb:8]: my vote is to move them to the facility helper.

**Review by kpethtel — COMMENTED** [review]
I love all the cleansing 🔥. I'm not sure I understand why we're renaming all the vcr cassettes. I do think our cassettes could use some organization and renaming, but I don't know why it's a part of this.

**Comment by Tonksthebear** [comment]
Happy to move it back if you'd prefer. We were deciphering which cassettes were still necessary and it was "off" that there is an `appointments/VAOS_v2` directory, and then everything else in `appointments` was a mixture of v0 and none v0 cassettes. It felt like the root `appointments` directory had no rhyme or reason so we added `creation` as an attempt to explain what the cassettes were for. As I said though, put a pin in it and revert the directory changes

File: modules/mobile/app/services/mobile/v0/appointments/proxy.rb [hunk 3]
```diff
@@ -6,152 +6,17 @@ module Mobile
...
-          if errors.size.positive?
-            Rails.logger.error('Mobile Appointments w/ Requests Error: ', errors: errors)
-            raise Common::Exceptions::BackendServiceException, 'MOBL_502_upstream_error'
-          end
-
-          { va: va_response, cc: cc_response, requests: requests_response }
-        end
-
-        def new_fetch_facilities(facility_ids, include_children)
+        def fetch_facilities(facility_ids, include_children)
```
Comment (kpethtel) [inline modules/mobile/app/services/mobile/v0/appointments/proxy.rb:19]: I'm not sure why we would still need this. It's a private function in a class that's going away, right?

File: modules/mobile/spec/request/appointments_create_request_spec.rb [hunk 4]
```diff
@@ -52,7 +52,8 @@
     it 'clears the cache' do
       expect(Mobile::V0::Appointment).to receive(:clear_cache).once
 
-      VCR.use_cassette('appointments/post_appointments_va_proposed_clinic_200', match_requests_on: %i[method uri]) do
+      VCR.use_cassette('appointments/post_appointments_va_proposed_clinic_200',
```
Comment (kpethtel) [inline modules/mobile/spec/request/appointments_create_request_spec.rb:55]: I would guess these changes are leftover from the path change you made. Might be worth changing them back just to avoid having a bunch of unnecessary changes in the PR.

**Review by kpethtel — COMMENTED** [review]
AFAICT this is correct. Andrew worked on the MFS thing more than I did (I think) so if he says we should get rid of the MFS facility stuff then I take his word on it. I think you need to finish deleting that proxy file but otherwise this seems good. I love to see all this code be deleted and especially all feature flag conditionals.

File: modules/mobile/config/initializers/statsd.rb (hunk 1 above)
Comment (Tonksthebear) [reply modules/mobile/config/initializers/statsd.rb:27]: @kpethtel Do you have any thoughts on this?

File: modules/mobile/spec/request/appointments_create_request_spec.rb (hunk 4 above)
Comment (Tonksthebear) [reply modules/mobile/spec/request/appointments_create_request_spec.rb:55]: It's actually rubocop. It won't pass without adding that newline so I think I'm out of luck

Comment (Tonksthebear) [reply modules/mobile/spec/request/appointments_create_request_spec.rb:55]: Interestingly though it only forces the newline on some files. It let me revert them on this file for instance but kept adding it back for others

File: modules/mobile/app/services/mobile/v0/appointments/proxy.rb (hunk 2 above)
Comment (Tonksthebear) [reply modules/mobile/app/services/mobile/v0/appointments/proxy.rb]: Moved it into the helper. lmk if I should change implementation

File: modules/mobile/spec/request/appointments_create_request_spec.rb (hunk 4 above)
Comment (kpethtel) [reply modules/mobile/spec/request/appointments_create_request_spec.rb]: Ok, that's cool. Just do whatever rubocop says.

File: modules/mobile/config/initializers/statsd.rb (hunk 1 above)
Comment (kpethtel) [reply modules/mobile/config/initializers/statsd.rb:27]: Yeah, i think andrew is probably right. It seems like the v2 metrics were overlooked somehow. We should probably do essentially redo these metrics for v2. Deleting these in this ticket seems fine, but I'd suggest adding the new ones in a separate ticket.

Comment (Tonksthebear) [reply modules/mobile/config/initializers/statsd.rb:27]: Ah it should be a separate ticket? I just pushed an update that adds it in this ticket because I figured it was "transitioning" from v0 to v2. We should be completely done with v0 to v2 after this

Comment (kpethtel) [reply modules/mobile/config/initializers/statsd.rb:27]: Just a suggestion. NBD.

Comment (Tonksthebear) [reply modules/mobile/config/initializers/statsd.rb:27]: Ok if you're cool I'd just like to change it here since it's the same lines of code changed and we can put a wrap on the v0 to v2 transition

File: modules/mobile/app/helpers/mobile/facilities_helper.rb [hunk 5]
```diff
@@ -1,9 +1,22 @@
...
+require 'lighthouse/facilities/client'
+
 module Mobile
   module FacilitiesHelper
     module_function
 
+    def fetch_facilities_from_ids(user, facility_ids, include_children)
+      ids = facility_ids.join(',')
+
+      facility_ids.each do |facility_id|
```
Comment (kpethtel) [inline modules/mobile/app/helpers/mobile/facilities_helper.rb:12]: Not really related to your PR, but I question the value of this logging. I'll probably file a ticket to do an audit of our logging and metrics soon so this can be handled at that time.

**Review by kpethtel — COMMENTED** [review]
I think this PR is ready to go live. I still think it's possible to break this apart into multiple smaller PRs but doing so doesn't get the main code removal under 500 lines so it's NBD. But removing VCR cassettes and initializing new metrics are not necessary to the code removal and could be follow-up PRs. But generally speaking, I do think there's more value in keeping unrelated parts of the change separate than there is in trying to get it all in a single PR.

**Review by kpethtel — APPROVED** [review]
LGTM 🚀

**Review by aherzberg — DISMISSED** [review]
LGTM

**Review by aherzberg — APPROVED** [review]
LGTM

**Review by kpethtel — APPROVED** [review]
🔥

**Review by rjohnson2011 — APPROVED** [review]
